src/logfile_values.py

```diff
@@ -7,42 +7,298 @@
         if ':' not in arg:
             raise ArgumentTypeError('Argument must have ":"')
-        self.name, column = arg.split(':', 1)
-        if not column.isdecimal():
-            raise ArgumentTypeError('Column must be a number')
-        self.column = int(column)
+        parts = arg.split(':')
+        if len(parts) != 4 and len(parts) != 2:
+            raise ValueError('Wrong number of options')
+
+        # Accepts 2 or 4 options. If 2 then function and period are NULL
+        parts += [None] * (4 - len(parts))
+        self.name, column, function, period = parts
+        if period:
+            pattern = re.compile("^\d+[A-Za-z]{1,3}$")
+            if not pattern.match(period):
+                raise ArgumentTypeError('Period must have number and unit')
+
+        self.column = column
+        self.function = function
+        self.period = period
+        # Here is container for metric values
+        self.values = []
+        self.last_value = 0
+        self.start_timestamp = 0
+        self.end_timestamp = 0
+        self.now = int(time.time())
+
+    def get_timeshift(self):
+        if self.period:
+            value = int(" ".join(re.findall("\d+", self.period.lower())))
+            unit = " ".join(re.findall("[a-z]+", self.period.lower()))
+            if unit == 's':
+                return timedelta(seconds=value).total_seconds()
+            if unit == 'min':
+                return timedelta(minutes=value).total_seconds()
+            if unit == 'h':
+                return timedelta(hours=value).total_seconds()
+            if unit == 'd':
+                return timedelta(days=value).total_seconds()
+        else:
+            return 0
+
+    def estimate_columns_value(self, fields):
+        '''
+        Apply some arithmetic on several columnsif needed
+        Warning: Estimates regardless of arithmetics rules        
+        '''
+        arr = [
+            s for s in self.column if s.isdigit() or s in ['/', '*', '+', '-']
+        ]
+        result = 0
+        for index, value in enumerate(arr):
+            if value.isdigit() and result == 0:
+                result += float(fields[int(value)])
+            elif value == '/':
+                result = result / float(fields[int(arr[index + 1])])
+            elif value == '*':
+                result = result * float(fields[int(arr[index + 1])])
+            elif value == '+':
+                result = result + float(fields[int(arr[index + 1])])
+            elif value == '-':
+                result = result - float(fields[int(arr[index + 1])])
+        return result
+
+    def get_duration(self):
+        '''Returns real duration when metric appeared in log file'''
+        duration = self.end_timestamp - self.start_timestamp
+        return duration
+
+    def get_median(self):
+        sorted_list = sorted(self.values)
+        i = len(sorted_list)
+        if not i % 2:
+            return (sorted_list[(i // 2) - 1] + sorted_list[i // 2]) / 2
```

Comment:
This looks wrong.  `//` would floor the value.  I think you need to use `i // 2` and `(i // 2) + 1`.

Reply:
I tested with function statistics.median. This variant looks correct

Reply:
True, I forgot to take into account that the list identifiers start from `0`.